pkgs/by-name/cc/cctools/meson.build

```diff
@@ -0,0 +1,602 @@
+# Build settings based on the upstream Xcode project.
+# See: https://github.com/apple-oss-distributions/cctools/blob/main/cctools.xcodeproj/project.pbxproj
```

Comment:
needing to keep the sources and perhaps build flags in synced with the meson and project.pbxproj seems like a possible pain point.

are there tools available to dump project.pbxproj data? being able to run `the-tool --project as --sources project.pbxproj` and `the-tool -project as --compiler-flags project.pbxproj` would be helpful for reviewers and future updates. not suggesting this needs to be written for this PR (or at all).

Reply:
Once cctools and ld64 are packaged, I’d just check the diff to see if any changes need to be made.

It’s not great, but dealing with xcbuild and Xcode project files is worse. I’ve wasted so much of my time on it with MoltenVK. I also tried using the project files for libiconv, and the dylibs built by xcbuild were all messed up. I have very little patience for dealing with it anymore, so my default now is not to even bother with them.

Reply:
ah right, looking at the diff of the pbxproj makes sense for updates. perhaps there could be some check added to `postUnpack` like.

```bash
# if the hash changed, the meson file might need to be updated with updated sources / compiler flags.
echo 'e5105b1abcf550a574c2dd9493b4481dd7e0587f2c830fe3cbb85f4d5aed56e3  ${src.name}/cctools.xcodeproj/project.pbxproj'|sha256sum -c
```

Reply:
I think I’ve code-golfed that enough. It now prints the hashes on a mismatch like a FOD. I also added the a Meson option for cctools to take a target prefix.